### ISSUE TITLE
cabal: p2p chat

### ISSUE THREAD

**Issue by hackergrrl** [dweb-camp-2019/projects#7]
**Cabal** is a peer-to-peer chat platform, not unlike apps like *Slack*, except it is open source, free, works offline, and doesn't require servers in order to be useful.

http://cabal.chat

![screenshot of cabal](https://cabal-club.github.io/desktop.png)

**Comment by benhylau** [comment]
- Does it need a bootstrap server when two users aren't on same LAN?
- What is a good way to exchange room key? QR codes?

**Comment by hackergrrl** [comment]
> Does it need a bootstrap server when two users aren't on same LAN?

Yes, of some kind. It can use a regular server that both sides know about, or bootstrap themselves to a DHT in order to locate each oher.

If needed, we could add some manual peering command, like `/addpeer 1.2.3.4` that could be remembered.

> What is a good way to exchange room key? QR codes?

QR codes would work (if implemented), or something easily passable by paper like https://github.com/Pringley/bitter

**Comment by RangerMauve** [comment]
I think we could share bootstrap servers for any dat-related projects.

**Comment by benhylau** [comment]
@noffle are you planning to host sessions on cabal or need space? If so please submit https://docs.google.com/forms/d/1Iet2xk6qwZZWkDHTEMCUn5zlb80IwXNtb6epk4-9O-s/ so we can start making arrangements. But I can also see this as a more unstructured thing (e.g. gather around in a tent and try this out / having binaries on a USB key that get passed around). Let me know what you prefer.

**Comment by hackergrrl** [comment]
Ben: something unstructured & informal is fine. :)
